clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -263,6 +264,15 @@ public FSDataOutputStream append(Path path, int i, Progressable progressable) th
     public boolean rename(Path src, Path dst) throws IOException {
         ObjectLocation srcObjectLoc = pathToObjectLocation(src);
         ObjectLocation dstObjectLoc = pathToObjectLocation(dst);
+        if (srcObjectLoc.getPath().isEmpty()) {
+            LOG.error("rename: src {} is root directory", src);
+            return false;
+        }
+        if (dstObjectLoc.getPath().isEmpty()) {
+            LOG.error("rename: dst {} is root directory", dst);
```

Comment:
And the same here: S3A doesn't allow me to move all files under `s3a://bucket/oops` to be under `s3a://bucket/`?

Reply:
As discussed, s3a does not allow renaming a directory when the destination is a non-empty directory. I'm planning to handle this general case at https://github.com/treeverse/lakeFS/issues/2061. 

This line mimics what s3a does in https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1501